crates/sui-types/src/temporary_store.rs

```diff
@@ -188,6 +189,9 @@ pub struct TemporaryStore<'backing> {
     /// Every package that was loaded from DB store during execution.
     /// These packages were not previously loaded into the temporary store.
     runtime_packages_loaded_from_db: RwLock<BTreeMap<ObjectID, Object>>,
+
+    // if this is populated the transaction execution will not run
+    formerly_deleted_shared_objects: DeletedSharedObjects,
```

Comment:
Maybe I'm just being dense, but when I first read this, I took it as "these shared objects were deleted but are not anymore."

Maybe `deleted_shared_input_objects` might be more direct?

Reply:
"already_deleted_shared_objects"?